source/com.android.billingclient/billing/Additions/Additions.cs

```diff
@@ -136,10 +140,12 @@ public Task<QueryProductDetailsResult> QueryProductDetailsAsync(QueryProductDeta
 
             QueryProductDetails(productDetailsParams, listener);
 
-            return tcs.Task;
+            var result = await tcs.Task;
+            GC.KeepAlive(listener);
+            return result;
```

Comment:
@tipa I think this is plausible that it could fix the `ObjectDisposedException`, but it seems like the old version would have had a similar issue.

The version number is unchanged, because we didn't ship it yet:
* [Xamarin.Android.Google.BillingClient.8.0.0.1.nupkg.zip](https://github.com/user-attachments/files/21440629/Xamarin.Android.Google.BillingClient.8.0.0.1.nupkg.zip)